### PR TITLE
Tabs: Added setActiveTab method

### PR DESCRIPTION
**Pull request by nilsburg**
Sets the first visible tab as active if not active is set 

| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes/no
| Fixed issues  | #186

### CODE REVIEW

File: Tabs.php [hunk 1]
```diff
@@ -154,7 +154,7 @@ protected function renderItems()
         $panes = [];
 
         if (!$this->hasActiveTab() && !empty($this->items)) {
-            $this->items[0]['active'] = true;
+            $this->setActiveTab();
```
Comment (samdark) [inline Tabs.php]: I'd rename it to `activateTab()` since `setActiveTab()` is a name for a setter.

File: Tabs.php [hunk 2]
```diff
@@ -290,4 +290,16 @@ public function renderPanes($panes)
     {
         return $this->renderTabContent ? "\n" . Html::tag('div', implode("\n", $panes), ['class' => 'tab-content']) : '';
     }
+    /**
+     * Sets the first visible tab as active
```
Comment (samdark) [inline Tabs.php]: Need `@since` tag.

File: Tabs.php [hunk 3]
```diff
@@ -290,4 +290,16 @@ public function renderPanes($panes)
...
+     */
+    protected function setActiveTab()
+    {
+        foreach ($this->items as $n => $item) {
+            if(!(isset($item['active']) && $item['active'] === false) && ((!isset($item['visible'])) || (isset($item['visible']) && $item['visible'] !== false))){
```
Comment (samdark) [inline Tabs.php]: The condition looks very complicated. It is possible to move it to separate variable and simplify it a bit?

File: Tabs.php (hunk 1 above)
Comment (nilsburg) [inline Tabs.php]: Yep, you are totally right.

File: Tabs.php (hunk 2 above)
Comment (nilsburg) [inline Tabs.php]: Added

File: Tabs.php (hunk 3 above)
Comment (nilsburg) [inline Tabs.php]: Ok, I've put the condition in two separate variables. Let me know if it is ok.

File: Tabs.php [hunk 4]
```diff
@@ -290,4 +290,19 @@ public function renderPanes($panes)
...
     }
+    /**
+     * Sets the first visible tab as active
+     * @since 2.0.7
+     */
+    protected function activateTab()
+    {
+        foreach ($this->items as $n => $item) {
+            $active = isset($item['active'])?$item['active']:null;
+            $visible = isset($item['visible'])?$item['visible']:null;
```
Comment (samdark) [inline Tabs.php]: $visible = ArrayHelper::getValue($item, 'visible', true);

File: Tabs.php [hunk 5]
```diff
@@ -290,4 +290,19 @@ public function renderPanes($panes)
...
         return $this->renderTabContent ? "\n" . Html::tag('div', implode("\n", $panes), ['class' => 'tab-content']) : '';
     }
+    /**
+     * Sets the first visible tab as active
+     * @since 2.0.7
+     */
+    protected function activateTab()
+    {
+        foreach ($this->items as $n => $item) {
+            $active = isset($item['active'])?$item['active']:null;
```
Comment (samdark) [inline Tabs.php]: $active = ArrayHelper::getValue($item, 'visible', null);

File: Tabs.php [hunk 6]
```diff
@@ -290,4 +290,19 @@ public function renderPanes($panes)
     {
         return $this->renderTabContent ? "\n" . Html::tag('div', implode("\n", $panes), ['class' => 'tab-content']) : '';
     }
+    /**
+     * Sets the first visible tab as active
+     * @since 2.0.7
+     */
+    protected function activateTab()
```
Comment (samdark) [inline Tabs.php]: Thinking about it more, I'd rename it to `activateFirstVisibleTab()`. More descriptive...

Comment (nilsburg) [inline Tabs.php]: No problem. Done

**Comment by nilsburg** [comment]
I've added two tests, one for checking the it actually activates the first _visible_ tab and another that checks that it activates the tab marked as active.

**Comment by samdark** [comment]
Everything's good now. Please add a line to changelog and I'll merge it.

**Comment by cebe** [comment]
Merged, thank you!

**Pull request closed without merging**